src/main/java/io/r2dbc/h2/H2Statement.java

```diff
@@ -102,6 +102,7 @@ public Flux<H2Result> execute() {
         Assert.requireTrue(!this.bindings.open, "No unfinished bindings!");
 
         return Flux.fromArray(this.sql.split(";"))
+            .filter(sql -> !sql.trim().isEmpty())
```

Comment:
I think there is a bigger problem.

`split(";")` cannot be used to break SQL into separate statements properly. This character may be used in comments, various types of character string literals, and quoted identifiers.

Reply:
You are right. If we remove the split, would it there be any serious implication? Mainly backward compatibility IMO.

Or we need to have a more sophisticated split than just string split.

Reply:
It would be so great to be able to make use of H2's internal parsing utils instead of us having to make a dance around H2 externally. Too bad, I wasn't able to convince @katzyn to bring this project closer to H2.

Reply:
Can we consider this fix as an interim patch while having much more careful changes to deal with `;` parsing?